File: BW8AGU/C#/Ch08/BW8AGU.Samples.CSharp/Common/TempDataHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.Storage;

namespace BW8AGU.Samples.CSharp.Common
{
    public static class TempDataHelper
    {
        public async static void SaveTempData(string data)
        {
            StorageFile tempFile = await ApplicationData.Current.TemporaryFolder.CreateFileAsync("temporary.txt", CreationCollisionOption.ReplaceExisting);
            await FileIO.WriteTextAsync(tempFile, data);
        }

        public async static Task<String> GetTempData()
        {
            try
            {
                StorageFile tempFile = await ApplicationData.Current.TemporaryFolder.GetFileAsync("temporary.txt");
                String content = await FileIO.ReadTextAsync(tempFile);
                return content;
            }
            catch
            {
                throw new Exception("Data not found!");
            }
        }
    }
}

File: BW8AGU/VB/Ch08/BW8AGU.Samples.VB/BW8AGU.Samples.VB/Common/TempDataHelper.vb
Public NotInheritable Class TempDataHelper
    Private Sub New()
    End Sub
    Public Shared Async Sub SaveTempData(data As String)
        Dim tempFile As StorageFile =
            Await ApplicationData.Current.TemporaryFolder.CreateFileAsync(
                "temporary.txt", CreationCollisionOption.ReplaceExisting)
        Await FileIO.WriteTextAsync(tempFile, data)
    End Sub

    Public Shared Async Function GetTempData() As Task(Of [String])
        Try
            Dim tempFile As StorageFile =
                Await ApplicationData.Current.TemporaryFolder.GetFileAsync("temporary.txt")
            Dim content As [String] = Await FileIO.ReadTextAsync(tempFile)
            Return content
        Catch
            Throw New Exception("Data not found!")
        End Try
    End Function
End Class

